go/porcelain/deploy.go

```diff
@@ -78,39 +77,25 @@ type uploadError struct {
 
 type FileBundle struct {
 	Name string
-	SHA  hash.Hash
+	Sum  string
 
+	// Path OR Buffer should be populated
+	Path   string
 	Buffer io.ReadSeeker
 }
 
-func (f *FileBundle) Sum() string {
-	return hex.EncodeToString(f.SHA.Sum(nil))
-}
-
 func (f *FileBundle) Read(p []byte) (n int, err error) {
 	return f.Buffer.Read(p)
 }
 
-func (f *FileBundle) Close() error {
-	return nil
+func (f *FileBundle) Seek(offset int64, whence int) (int64, error) {
+	return f.Buffer.Seek(offset, whence)
```

Comment:
doesn't this have the problem that the Buffer could not be populated?

Reply:
Yes. I can add a check, but it would be programmer error if you attempted to use this interface without Buffer populated.

I could think about a safer way to do this, but it would be more changes.